### PR TITLE
doc: add information on how to run test cases

### PR DESCRIPTION
**Pull request by pieleric**

### CODE REVIEW

File: doc/develop/extending.rst [hunk 1]
```diff
@@ -619,6 +619,27 @@ always run from within the main GUI thread. Another way is to call every GUI
 related function using the special ``wx.CallAfter()`` function.
 
 
+Running test cases
+==================
+The source code comes with a large set of unit tests and some integration tests.
+They allow check the behaviour of the different parts of Odemis.
```
Comment (Sabrina85) [inline doc/develop/extending.rst]: typo: They allow checking...

File: doc/develop/extending.rst [hunk 2]
```diff
@@ -619,6 +619,27 @@ always run from within the main GUI thread. Another way is to call every GUI
...
+After changes are made to the source, the tests should be rerun in order to validate
+these changes. To run the test cases, it is recommended to first create a directory
+next to the odemis directory, and name it ``odemis-testing``. You can also have
+another directory ``mic-odm-yaml``, which contains all the microscopes files.
```
Comment (Sabrina85) [inline doc/develop/extending.rst]: odemis-testing: empty directory or copy of the whole odemis folder?
mic-odm-yaml: i have this folder, which includes a lot of subfolders with yaml files. What is the purpose of having this folder or does the testing need all these yaml files? Sounds like it is an optional step?

File: doc/develop/extending.rst [hunk 3]
```diff
@@ -619,6 +619,27 @@ always run from within the main GUI thread. Another way is to call every GUI
...
+It is then possible to run all the test cases with this command::
+
```
Comment (Sabrina85) [inline doc/develop/extending.rst]: maybe add which folder the command needs to be run from..

**Comment by pieleric** [comment]
I've updated the text with the recommended changes. Please merge if that now looks fine.

**Pull request merged**